### PR TITLE
[DEVOPS-1566] - add DCT test workflow

### PR DESCRIPTION
**Pull request by Eeebru**
- add DCT test workflow

### CODE REVIEW

File: .github/workflows/DCT-test.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        with:
+          azure-creds: ${{ secrets.AZURE_KV_CI_SERVICE_PRINCIPAL }}
+          azure-keyvault-name: "bitwarden-ci"
+
+      - name: Pull image from ACR
+        run: docker pull $_AZ_REGISTRY/$_PROJECT_NAME:$_IMAGE_TAG
+
+      - name: Tag DH image
+        run: |
+          docker tag $_AZ_REGISTRY/$_PROJECT_NAME:$_IMAGE_TAG bitwarden/$_PROJECT_NAME:$_IMAGE_TAG
```
Comment (mimartin12) [inline .github/workflows/DCT-test.yml:54]:
```suggestion
        run: docker tag $_AZ_REGISTRY/$_PROJECT_NAME:$_IMAGE_TAG bitwarden/$_PROJECT_NAME:$_IMAGE_TAG
```
🎨 
Single line for the `run` command, or maybe consider merging this step with the one above.

File: .github/workflows/DCT-test.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+      - name: Push image to Docker Hub
+        env:
+          DOCKER_CONTENT_TRUST: 1
+          DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE: ${{ steps.setup-dct.outputs.dct-delegate-repo-passphrase }}
+        run: |
+          docker push bitwarden/$_PROJECT_NAME:$_IMAGE_TAG
```
Comment (mimartin12) [inline .github/workflows/DCT-test.yml:61]: 🎨  single line for `run` command.
```suggestion
        run: docker push bitwarden/$_PROJECT_NAME:$_IMAGE_TAG
```

**Review by mimartin12 — CHANGES_REQUESTED** [review]

**Review by vgrassia — COMMENTED** [review]
What is the purpose of this workflow?  Do we need the ability to push any image manually to Docker Hub?

**Comment by Eeebru** [comment]
> What is the purpose of this workflow? Do we need the ability to push any image manually to Docker Hub?

This is only used when we create a new repository on DH to test if the signing is working.

**Comment by vgrassia** [comment]
To make this more secure, can the workflow have all inputs removed and pinned for `mssqlmigratorutility` and then move this whole workflow over to the `devops` repository? I can see this being an issue if someone runs the workflow with the wrong image/tag and deploys a signed image to one of our Docker Hub repositories.

**Comment by Eeebru** [comment]
> To make this more secure, can the workflow have all inputs removed and pinned for `mssqlmigratorutility` and then move this whole workflow over to the `devops` repository? I can see this being an issue if someone runs the workflow with the wrong image/tag and deploys a signed image to one of our Docker Hub repositories.

Okay, will update the workflow reference only `mssqlmigratorutility`, however, Joseph suggested the workflow to be here. We can have the conversation on our Team sync.

**Comment by Eeebru** [comment]
Hello @vgrassia, at the team sync yesterday, Joseph said he would leave me to make the decision of where this test wf should be. I am of the opinion that self-host is the only place where we are pushing any images to DH, so it makes sense if we leave this workflow in this repo, instead of DevOps repo. It will also be pretty redundant, since it would only be run when we add repos to DH, leaving it in the DevOps repo means we might delete it along the line when it's not being used.

File: .github/workflows/DCT-test.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Login to Azure ACR
+        run: az acr login -n ${_AZ_REGISTRY%.azurecr.io}
+
+      - name: Setup project name
+        id: setup
+        run: |
+          PROJECT_NAME=$(echo "${{ env._REPO }}" | awk '{print tolower($0)}')
+          echo "PROJECT_NAME: $PROJECT_NAME"
+          echo "_PROJECT_NAME=$PROJECT_NAME" >> $GITHUB_ENV
+
```
Comment (vgrassia) [inline .github/workflows/DCT-test.yml:33]:
```suggestion
```

File: .github/workflows/DCT-test.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        id: setup-dct
+        uses: bitwarden/gh-actions/setup-docker-trust@082f5e05ed97c3601c6f3179250b1a761c4d647f
+        with:
+          azure-creds: ${{ secrets.AZURE_KV_CI_SERVICE_PRINCIPAL }}
+          azure-keyvault-name: "bitwarden-ci"
+
+      - name: Pull and tag image from ACR
+        run: |
+          docker pull $_AZ_REGISTRY/$_PROJECT_NAME:$_IMAGE_TAG
+          docker tag $_AZ_REGISTRY/$_PROJECT_NAME:$_IMAGE_TAG bitwarden/$_PROJECT_NAME:$_IMAGE_TAG
```
Comment (vgrassia) [inline .github/workflows/DCT-test.yml:45]:
```suggestion
          docker pull $_AZ_REGISTRY/$_REPO:$_IMAGE_TAG
          docker tag $_AZ_REGISTRY/$_REPO:$_IMAGE_TAG bitwarden/$_REPO:$_IMAGE_TAG
```

File: .github/workflows/DCT-test.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+      - name: Push image to Docker Hub
+        env:
+          DOCKER_CONTENT_TRUST: 1
+          DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE: ${{ steps.setup-dct.outputs.dct-delegate-repo-passphrase }}
+        run: docker push bitwarden/$_PROJECT_NAME:$_IMAGE_TAG
```
Comment (vgrassia) [inline .github/workflows/DCT-test.yml:51]:
```suggestion
        run: docker push bitwarden/$_REPO:$_IMAGE_TAG
```

**Review by vgrassia — CHANGES_REQUESTED** [review]

**Review by vgrassia — APPROVED** [review]

**Review by michalchecinski — APPROVED** [review]

**Review by mimartin12 — APPROVED** [review]

**Pull request merged**